pkg/vm/stackitem/json.go

```diff
@@ -0,0 +1,202 @@
+package stackitem
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	gio "io"
+	"math"
+	"math/big"
+
+	"github.com/nspcc-dev/neo-go/pkg/io"
+)
+
+// decoder is a wrapper around json.Decoder helping to mimic C# json decoder behaviour.
+type decoder struct {
+	json.Decoder
+
+	depth int
+}
+
+// MaxAllowedInteger is the maximum integer allowed to be encoded.
+const MaxAllowedInteger = 2<<53 - 1
+
+// maxJSONDepth is a maximum allowed depth-level of decoded JSON.
+const maxJSONDepth = 10
+
+// ToJSON encodes Item to JSON.
+// It behaves as following:
+//   ByteArray -> base64 string
+//   BigInteger -> number
+//   Bool -> bool
+//   Null -> null
+//   Array -> array
+//   Map -> map with keys as UTF-8 bytes
```

Comment:
I think `Struct` should work just fine too, although missing support for `Buffer` seems to be suspicious.

Reply:
Yep, see neo-project/neo#1697. Maybe we can even add this as an extension for now.

Reply:
As for structs, try running `11c64a1017d041248d264b` vs `11c34a1017d041248d264b`.

Reply:
`Struct` support is here, just wasn't mentioned in comments.
Added `Buffer` support.

Reply:
>  Added Buffer support.

Com-pa-ti-bi-li-ty.

0.90.0 is to be compatible with preview2, we absolutely need this stable known reference that anyone trying to use Neo 3 now understands. I'm fine with a draft PR for it, but it's not appropriate to add it as of now, especially given that the upstream bug is not solved in any way yet.